core/src/main/scala/kafka/producer/BaseProducer.scala

```diff
@@ -19,19 +19,21 @@ package kafka.producer
 
 import java.util.Properties
 
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+
 // A base producer used whenever we need to have options for both old and new producers;
 // this class will be removed once we fully rolled out 0.9
 @deprecated("This trait has been deprecated and will be removed in a future release. " +
             "Please use org.apache.kafka.clients.producer.KafkaProducer instead.", "0.10.0.0")
 trait BaseProducer {
   def send(topic: String, key: Array[Byte], value: Array[Byte])
+  def send(record: ProducerRecord[Array[Byte], Array[Byte]])
```

Comment:
Should we just get rid of the original form of this method from the interface? There's only one other use of the trait's method (as opposed to `OldProducer`'s own implementation which could be made `private`) and it'd be trivial to convert it to the new format.

@ijuma it looks like you had marked this as deprecated, which sort of implies public API, but afaik this is not actually public API. Is it safe to modify? Was this just marked w/ deprecated in order to make sure it gets cleaned up with the rest of the old producer stuff?

Reply:
I marked it as deprecated for two reasons:

1. As you said, so that we remove it with the old producer.
2. Avoid deprecation warnings as much as possible. Deprecated classes that use other deprecated classes don't trigger a warning.

Reply:
Ok, so it sounds safe to modify then.